analyzers/tests/SonarAnalyzer.UnitTest/Rules/CastShouldNotBeDuplicatedTest.cs

```diff
@@ -25,18 +25,34 @@ namespace SonarAnalyzer.UnitTest.Rules
     [TestClass]
     public class CastShouldNotBeDuplicatedTest
     {
+        private readonly VerifierBuilder builder = new VerifierBuilder<CastShouldNotBeDuplicated>();
+
         [TestMethod]
         public void CastShouldNotBeDuplicated() =>
-            OldVerifier.VerifyAnalyzer(@"TestCases\CastShouldNotBeDuplicated.cs", new CastShouldNotBeDuplicated());
+            builder.AddPaths("CastShouldNotBeDuplicated.cs").Verify();
 
 #if NET
+
         [TestMethod]
         public void CastShouldNotBeDuplicated_CSharp9() =>
-            OldVerifier.VerifyAnalyzerFromCSharp9Library(@"TestCases\CastShouldNotBeDuplicated.CSharp9.cs", new CastShouldNotBeDuplicated());
+            builder.AddPaths("CastShouldNotBeDuplicated.CSharp9.cs")
+                .WithOptions(ParseOptionsHelper.FromCSharp9)
+                .Verify();
 
         [TestMethod]
         public void CastShouldNotBeDuplicated_CSharp10() =>
-            OldVerifier.VerifyAnalyzerFromCSharp10Library(@"TestCases\CastShouldNotBeDuplicated.CSharp10.cs", new CastShouldNotBeDuplicated());
+            builder.AddPaths("CastShouldNotBeDuplicated.CSharp10.cs")
+                .WithOptions(ParseOptionsHelper.FromCSharp10)
+                .Verify();
+
+        [TestMethod]
+        public void CastShouldNotBeDuplicated_CSharp11() =>
+            builder.AddPaths("CastShouldNotBeDuplicated.CSharp11.cs")
+                .WithTopLevelStatements()
```

Comment:
void ListPattern() can go into a class and you will not need the WithTopLevelStatements.